homeassistant/components/hunterdouglas_powerview/button.py

```diff
@@ -0,0 +1,126 @@
+"""Support for Powerview advanced features."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aiopvapi.resources.shade import BaseShade, factory as PvShade
+
+from homeassistant.components.button import ButtonDeviceClass, ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    COORDINATOR,
+    DEVICE_INFO,
+    DOMAIN,
+    PV_API,
+    PV_ROOM_DATA,
+    PV_SHADE_DATA,
+    ROOM_ID_IN_SHADE,
+    ROOM_NAME_UNICODE,
+)
+from .coordinator import PowerviewShadeUpdateCoordinator
+from .entity import ShadeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the hunter douglas advanced feature buttons."""
+
+    pv_data = hass.data[DOMAIN][entry.entry_id]
+    room_data: dict[str | int, Any] = pv_data[PV_ROOM_DATA]
+    shade_data = pv_data[PV_SHADE_DATA]
+    pv_request = pv_data[PV_API]
+    coordinator: PowerviewShadeUpdateCoordinator = pv_data[COORDINATOR]
+    device_info: dict[str, Any] = pv_data[DEVICE_INFO]
+
+    entities: list[ButtonEntity] = []
+    for raw_shade in shade_data.values():
+        shade: BaseShade = PvShade(raw_shade, pv_request)
+        name_before_refresh = shade.name
+        room_id = shade.raw_data.get(ROOM_ID_IN_SHADE)
+        room_name = room_data.get(room_id, {}).get(ROOM_NAME_UNICODE, "")
+
+        entities.append(
+            ButtonCalibrate(
```

Comment:
These are all very similar models. We should use ButtonEntityDescription for these

Reply:
Example here
https://github.com/home-assistant/core/blob/dev/homeassistant/components/bond/button.py#L28

Reply:
done